<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MyMainWindow</class>
 <widget class="QMainWindow" name="MyMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>657</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>OpenGLDemo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGLWidget" name="glwidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>900</width>
      <height>600</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>300</width>
      <height>300</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
   </widget>
   <widget class="QTabWidget" name="Tabs">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>10</y>
      <width>261</width>
      <height>611</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>3</number>
    </property>
    <widget class="QWidget" name="lists">
     <attribute name="title">
      <string>Lists</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>241</width>
        <height>171</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Faces</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="faceList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy">
          <enum>Qt::ClickFocus</enum>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::ExtendedSelection</enum>
         </property>
         <property name="layoutMode">
          <enum>QListView::SinglePass</enum>
         </property>
         <property name="sortingEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>241</width>
        <height>181</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Edges</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="edgesList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy">
          <enum>Qt::ClickFocus</enum>
         </property>
         <property name="sortingEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>380</y>
        <width>241</width>
        <height>191</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Vertices</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="vertexList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy">
          <enum>Qt::ClickFocus</enum>
         </property>
         <property name="sortingEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="mesh_tab">
     <attribute name="title">
      <string>Mesh</string>
     </attribute>
     <widget class="QGroupBox" name="MeshOperations">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>201</width>
        <height>141</height>
       </rect>
      </property>
      <property name="title">
       <string>Mesh Operations</string>
      </property>
      <widget class="QPushButton" name="triangulate">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>81</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Split 
Quad</string>
       </property>
      </widget>
      <widget class="QPushButton" name="deleteVertexButton">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>80</y>
         <width>81</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Delete 
Vertex</string>
       </property>
      </widget>
      <widget class="QPushButton" name="addEdgeButton">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>30</y>
         <width>81</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Add 
Edge</string>
       </property>
      </widget>
      <widget class="QPushButton" name="addVertexButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>81</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Add 
Vertex</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="MoveVertex">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>160</y>
        <width>201</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>Move Vertex</string>
      </property>
      <widget class="QDoubleSpinBox" name="moveVertexX">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>91</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-5.000000000000000</double>
       </property>
       <property name="maximum">
        <double>5.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="moveVertexY">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>60</y>
         <width>91</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-5.000000000000000</double>
       </property>
       <property name="maximum">
        <double>5.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="moveVertexZ">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>90</y>
         <width>91</width>
         <height>24</height>
        </rect>
       </property>
       <property name="minimum">
        <double>-5.000000000000000</double>
       </property>
       <property name="maximum">
        <double>5.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.500000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>X:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Y:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>90</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Z:</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="FaceColor">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>300</y>
        <width>201</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>Change Face Color</string>
      </property>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>R:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>G:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>90</y>
         <width>62</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>B:</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="faceColorRed">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>30</y>
         <width>91</width>
         <height>24</height>
        </rect>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="faceColorGreen">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>60</y>
         <width>91</width>
         <height>24</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="faceColorBlue">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>90</y>
         <width>91</width>
         <height>24</height>
        </rect>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="subdivideButton">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>430</y>
        <width>121</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Smooth Mesh</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>480</y>
        <width>201</width>
        <height>91</height>
       </rect>
      </property>
      <property name="title">
       <string>Default Meshes</string>
      </property>
      <widget class="QPushButton" name="planeButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>81</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Plane</string>
       </property>
      </widget>
      <widget class="QPushButton" name="cubeButton">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>30</y>
         <width>81</width>
         <height>51</height>
        </rect>
       </property>
       <property name="text">
        <string>Cube</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Cam</string>
     </attribute>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Camera View:</string>
      </property>
     </widget>
     <widget class="QComboBox" name="cameraViewComboBox">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>40</y>
        <width>121</width>
        <height>26</height>
       </rect>
      </property>
      <property name="currentText">
       <string/>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>90</y>
        <width>181</width>
        <height>131</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Snap Camera to Selected:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="SnapVertex">
         <property name="text">
          <string>Vertex</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="snapEdge">
         <property name="text">
          <string>Edge</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="Face">
         <property name="text">
          <string>Face</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="Deformer">
     <attribute name="title">
      <string>Deform</string>
     </attribute>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>10</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Upload Object</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>90</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="moveCPY">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>500</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="moveCPX">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>500</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QSpinBox" name="numLatticeZ">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>110</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>2</number>
      </property>
      <property name="maximum">
       <number>5</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="numLatticeX">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>2</number>
      </property>
      <property name="maximum">
       <number>5</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>90</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_2">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>40</y>
        <width>121</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Export Object</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>231</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Free-Form Deformation Lattice</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>480</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>450</y>
        <width>151</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Move Control Point</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QListWidget" name="latticeList">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>220</y>
        <width>191</width>
        <height>221</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>480</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>90</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="numLatticeY">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>110</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>2</number>
      </property>
      <property name="maximum">
       <number>5</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>480</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="createLattice">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>191</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Create Lattice</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="moveCPZ">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>500</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <double>-99.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>180</y>
        <width>191</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Control Points</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox">
      <property name="geometry">
       <rect>
        <x>101</x>
        <y>550</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>530</y>
        <width>46</width>
        <height>13</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Twist</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QWidget</extends>
   <header>myglwidget.h</header>
   <container>1</container>
   <slots>
    <signal>sendRoot(QTreeWidgetItem*)</signal>
    <signal>sendFaceItem(QListWidgetItem*)</signal>
    <signal>sendVertexItem(QListWidgetItem*)</signal>
    <signal>sendHalfEdgeItem(QListWidgetItem*)</signal>
    <signal>clearLists()</signal>
    <signal>sendFace(QListWidgetItem*)</signal>
    <signal>sort()</signal>
    <signal>addEdge(int)</signal>
    <signal>addVertex(int)</signal>
    <signal>addFace(int)</signal>
    <signal>clearFaceList()</signal>
    <signal>signal1()</signal>
    <signal>deleteFace()</signal>
    <signal>clearEdgeList()</signal>
    <signal>clearVertexList()</signal>
    <slot>MakeMagic()</slot>
    <slot>addNode()</slot>
    <slot>reDraw()</slot>
    <slot>clearGraph()</slot>
    <slot>spin()</slot>
    <slot>receiveFace(Face*)</slot>
    <slot>receiveFaces(Face*,Face*)</slot>
    <slot>splitQuad(Face*)</slot>
    <slot>sharedVertexToEdge()</slot>
    <slot>receiveVertex(Vertex*)</slot>
    <slot>addVertex(HalfEdge*)</slot>
    <slot>receiveHalfEdge(HalfEdge*)</slot>
    <slot>deleteVertex()</slot>
    <slot>zoomIn()</slot>
    <slot>zoomOut()</slot>
    <slot>horizontalRotate(int)</slot>
    <slot>verticalRotate(int)</slot>
    <slot>cubeSpeed(int)</slot>
    <slot>cubeDirectionReversed()</slot>
    <slot>moveVertex_X(QString)</slot>
    <slot>moveVertex_Y(QString)</slot>
    <slot>moveVertex_Z(QString)</slot>
    <slot>colorRed(double)</slot>
    <slot>colorBlue(double)</slot>
    <slot>colorGreen(double)</slot>
    <slot>drawCharacter()</slot>
    <slot>draw()</slot>
    <slot>receiveRoot(QTreeWidgetItem*)</slot>
    <slot>keyframeChanged(int)</slot>
    <slot>numFramesEntered(QString)</slot>
    <slot>startAnimation()</slot>
    <slot>frameChangedToKeyframe()</slot>
    <slot>frameSliderMoved(int)</slot>
    <slot>slot1()</slot>
    <slot>faceColorChanged(int,int,double)</slot>
    <slot>vertexMoved(int,int,double)</slot>
    <slot>currentVertex(int)</slot>
    <slot>currentFace(int)</slot>
    <slot>currentEdge(int)</slot>
    <slot>splitQuad()</slot>
    <slot>addEdge()</slot>
    <slot>addVertex()</slot>
    <slot>planarityTest()</slot>
    <slot>currentEdge(QString)</slot>
    <slot>currentFace(QString)</slot>
    <slot>currentVertex(QString)</slot>
    <slot>uploadObj()</slot>
    <slot>exportObj()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>glwidget</receiver>
   <slot>uploadObj()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1002</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>71</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>glwidget</receiver>
   <slot>exportObj()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1036</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>862</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
